com.hazelcast.jet.pipeline.SourceBuilder_TopologyChangeTest.test_restartJob_nodeTerminated [HZ-1394]

_master_ (commit 98551638971b0badced3013beea408f4d68b16e1)

Failed on OracleJDK17: https://jenkins.hazelcast.com/view/Official%20Builds/job/Hazelcast-master-OracleJDK17/41/testReport/com.hazelcast.jet.pipeline/SourceBuilder_TopologyChangeTest/test_restartJob_nodeTerminated/

<details><summary>Stacktrace:</summary>

```
java.lang.AssertionError: expected:<100> but was:<101>
	at org.junit.Assert.fail(Assert.java:89)
	at org.junit.Assert.failNotEquals(Assert.java:835)
	at org.junit.Assert.assertEquals(Assert.java:647)
	at org.junit.Assert.assertEquals(Assert.java:633)
	at com.hazelcast.jet.pipeline.SourceBuilder_TopologyChangeTest.testTopologyChange(SourceBuilder_TopologyChangeTest.java:135)
	at com.hazelcast.jet.pipeline.SourceBuilder_TopologyChangeTest.test_restartJob_nodeTerminated(SourceBuilder_TopologyChangeTest.java:62)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:115)
	at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:107)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.lang.Thread.run(Thread.java:833)
```

</details>

Standard output can be downloaded from https://console.aws.amazon.com/s3/buckets/j-artifacts/Hazelcast-master-OracleJDK17/41/

--- a/hazelcast/src/main/java/com/hazelcast/jet/impl/execution/ConcurrentInboundEdgeStream.java
+++ b/hazelcast/src/main/java/com/hazelcast/jet/impl/execution/ConcurrentInboundEdgeStream.java
@@ -221,6 +221,7 @@ public final class ConcurrentInboundEdgeStream {
                     specialItemsStash.add(currentBarrier);
                     currentBarrier = null;
                     receivedBarriers.clear();
+                    break;
                 }
             }
 

--- a/hazelcast/src/test/java/com/hazelcast/jet/impl/execution/ConcurrentInboundEdgeStreamTest.java
+++ b/hazelcast/src/test/java/com/hazelcast/jet/impl/execution/ConcurrentInboundEdgeStreamTest.java
@@ -194,7 +194,8 @@ public class ConcurrentInboundEdgeStreamTest {
         add(q2, wm(1));
 
         // Then
-        drainAndAssert(MADE_PROGRESS, barrier(0), wm(1));
+        drainAndAssert(MADE_PROGRESS, barrier(0));
+        drainAndAssert(MADE_PROGRESS, wm(1));
     }
 
     @Test
@@ -263,8 +264,8 @@ public class ConcurrentInboundEdgeStreamTest {
 
         add(q1, barrier(1));
         add(q2, 1);
-        drainAndAssert(MADE_PROGRESS, 1);
         drainAndAssert(MADE_PROGRESS, barrier(1));
+        drainAndAssert(MADE_PROGRESS, 1);
     }
 
     private void drainAndAssert(ProgressState expectedState, Object... expectedItems) {

